### ISSUE TITLE
Update CI workflows

### ISSUE THREAD

**Issue by ygrishajev** [akash-network/console#144]
Currently 2 workflows are happening in parallel on merge to `main`:
1) tests
2) release
![image](https://github.com/akash-network/cloudmos/assets/10720827/5de41539-5554-4b75-9109-159213de3c33)

which doesn't make much sense as release should depend on tests to be cancelled if ones fail. There are ways to do it a more logical fashion. Some of them are:
- protect master and run test only on feature branches. So that in master it only releases. This one has an issue that tests wouldn't run if a hot-fix is required directly in `main`
- run release when tests are green. this one is more safe but would double run tests for the same commit if merge is done with a rebase.

It looks like option 2 would be more consistent though.
Some optimisation can be done there by finding/creating a custom action to skip tests if those are already run against `head_commit`.

**Tasks:**

- [ ] Make sure release does depend on tests
- [ ] Skip running tests if already done for current `head_commit`
